lane.py

```diff
@@ -387,897 +372,796 @@ def has_next_page(self):
+    # "Fiction".
+    identifier = Column(Unicode)
+
+    # display_name is the name of the lane as shown to patrons.  It's
+    # okay for this to be duplicated across the library, but it's not
+    # okay to have two lanes with the same parent and the same display
+    # name -- that would be confusing.
+    display_name = Column(Unicode)
+
+    # True = Fiction only
+    # False = Nonfiction only
+    # null = Both fiction and nonfiction
+    #
+    # This may interact with lane_genres, for genres such as Humor
+    # which can apply to either fiction or nonfiction.
+    fiction = Column(Boolean, index=True, nullable=True)
+
+    # A lane may be restricted to works classified for specific audiences
+    # (e.g. only Young Adult works).
+    _audiences = Column(ARRAY(Unicode), name='audiences')
+
+    # A lane may further be restricted to works classified as suitable
+    # for a specific age range.
+    _target_age = Column(INT4RANGE, name="target_age", index=True)
+
+    # A lane may be restricted to works available in certain languages.
+    languages = Column(ARRAY(Unicode))
+
+    # A lane may be restricted to works in certain media (e.g. only
+    # audiobooks).
+    media = Column(ARRAY(Unicode))
+
+    # TODO: At some point it may be possible to restrict a lane to certain
+    # formats (e.g. only electronic materials or only codices).
+
+    # Only books licensed through this DataSource will be shown.
+    license_datasource_id = Column(
+        Integer, ForeignKey('datasources.id'), index=True,
+        nullable=True
+    )
 
-        Works will:
+    # Only books on one or more CustomLists obtained from this
+    # DataSource will be shown.
+    _list_datasource_id = Column(
+        Integer, ForeignKey('datasources.id'), index=True,
+        nullable=True
+    )
 
-        * Be in one of the languages listed in `languages`,
-          and not one of the languages listed in `exclude_languages`.
+    # Only the books on these specific CustomLists will be shown.
+    customlists = relationship(
+        "CustomList", secondary=lambda: lanes_customlists,
+        backref="lane"
+    )
 
-        * Be filed under of the genres listed in `self.genre_ids` (or, if
-          `self.include_subgenres` is True, any of those genres'
-          subgenres).
+    # This has no effect unless list_datasource_id or
+    # list_identifier_id is also set. If this is set, then a book will
+    # only be shown if it was seen on an appropriate list within this
+    # number of days. If the number is zero, the book must be
+    # _currently_ on an appropriate list.
+    list_seen_in_previous_days = Column(Integer, nullable=True)
+
+    # If this is set to True, then a book will show up in a lane only
+    # if it would _also_ show up in its parent lane.
+    #
+    # Currently this has no effect unless list_datasource_id or
+    # list_identifier_id is also set.
+    inherit_parent_restrictions = Column(Boolean, default=False, nullable=False)
```

Comment:
I'd like to discuss this option tomorrow.

Reply:
Sounds good. I believe the "currently this has no effect" comment is wrong so I took it out.